pkg/operator/crmclusterstop_v1.go

```diff
@@ -0,0 +1,197 @@
+		baseOperator: newBaseOperator(
+			CrmClusterStopOperatorName, operationID, arguments, options.BaseOperatorOptions...,
+		),
+		clusterClient: cluster.NewDefaultClusterClient(),
+		// wait before each execution: 0s, 1.5s, 4.5s, 13.5s, 40.5s
+		retryOptions: support.BackoffOptions{
+			InitialDelay: 500 * time.Millisecond,
+			MaxDelay:     1 * time.Minute,
+			MaxRetries:   5,
+			Factor:       3,
+		},
+	}
+
+	for _, opt := range options.OperatorOptions {
+		opt(crmClusterStop)
+	}
+
+	return &Executor{
+		phaser:      crmClusterStop,
+		operationID: operationID,
+		logger:      crmClusterStop.logger,
+	}
+}
+
+func (c *CrmClusterStop) plan(ctx context.Context) (bool, error) {
+	// check if the cluster is not started.
+	isOnline := c.clusterClient.IsHostOnline(ctx)
+	c.resources[beforeDiffField] = !isOnline
+
+	if !isOnline {
+		c.logger.Info("CRM cluster is not online, skipping stop operation")
+		c.resources[afterDiffField] = true
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (c *CrmClusterStop) commit(ctx context.Context) error {
+	// If the cluster is not idle, we cannot stop it safely.
+	err := c.ensureIsIdle(ctx)
+	if err != nil {
+		return fmt.Errorf("cluster is not in IDLE state, cannot stop: %w", err)
+	}
+
+	result := <-support.AsyncExponentialBackoff(
+		ctx,
+		c.retryOptions,
+		func() (bool, error) {
+			return true, c.clusterClient.StopCluster(ctx)
+		},
+	)
+
+	if result.Err != nil {
+		return fmt.Errorf("error stopping CRM cluster: %w", result.Err)
+	}
+
+	return nil
+}
+
+func (c *CrmClusterStop) rollback(ctx context.Context) error {
+	// We can start the cluster again if it was stopped successfully.
+	return c.clusterClient.StartCluster(ctx)
+}
+
+func (c *CrmClusterStop) verify(ctx context.Context) error {
+	result := <-support.AsyncExponentialBackoff(
+		ctx,
+		c.retryOptions,
+		func() (bool, error) {
+			isOnline := c.clusterClient.IsHostOnline(ctx)
+			if isOnline {
+				return false, fmt.Errorf("CRM cluster is still online, expected offline state")
+			}
+			return true, nil
+		},
+	)
+
+	if result.Err != nil {
+		return result.Err
+	}
+
+	c.resources[afterDiffField] = true
+	return nil
+}
+
+func (c *CrmClusterStop) operationDiff(ctx context.Context) map[string]any {
+	diff := make(map[string]any)
+
+	beforeDiffOutput := CrmClusterStopDiffOutput{
+		Stopped: c.resources[beforeDiffField].(bool),
+	}
+	before, _ := json.Marshal(beforeDiffOutput)
+	diff["before"] = string(before)
+
+	afterDiffOutput := CrmClusterStopDiffOutput{
+		Stopped: c.resources[afterDiffField].(bool),
+	}
+	after, _ := json.Marshal(afterDiffOutput)
+	diff["after"] = string(after)
+
+	return diff
+}
+
+// Ensure the CRM cluster is idle before proceeding with the operation.
+// This is a safety check to ensure that the cluster is in a stable state.
+// If the cluster is not idle, we will retry until it becomes idle or the maximum retries are reached.
+func (c *CrmClusterStop) ensureIsIdle(ctx context.Context) error {
```

Comment:
nitpick:
now I see this we could most probably make this a objectless function passing the retry options and cluster client as params, to reuse the same function in the start/stop.